More data in Undermine

pepedressingroom opened the pull request:
add ns.requirement.Quest support for checking "completed on account"
add reputation quest ids of rares
add paragon reputation rewards
add C.H.E.T.T. List
add Shipping & Handling rewards
add a node "unlock skyrocketing races world quests"
add D.R.I.V.E. customization's checklist

pepedressingroom committed:
Quest requirement

support for completed on account

pepedressingroom committed:
more data

pepedressingroom committed:
Merge remote-tracking branch 'upstream/master' into undermine

pepedressingroom committed:
locale

pepedressingroom committed:
zhTW

pepedressingroom committed:
C.H.E.T.T.

pepedressingroom committed:
Merge remote-tracking branch 'upstream/master' into undermine

adavak committed:
lint

Krovikan commented:
Lint fails points to `_` in these lines of undermine.lua:

572 requires = ns.requirement.Quest(83542, `_`, `_`, true), -- ![Quick Gills for Gold Now] (warband)

727 requires = ns.requirement.Quest(86961, `_`, `_`, true), -- ![Diversified Investments] (warband)

817 ns.requirement.Quest(87374, `_`, `_`, true) -- ![C.H.E.T.T.ing In] (warband)

837 ns.requirement.Quest(87374, `_`, `_`, true) -- ![C.H.E.T.T.ing In] (warband)

pepedressingroom committed:
fix lint

pepedressingroom committed:
Merge remote-tracking branch 'origin/undermine' into undermine

pepedressingroom committed:
Merge remote-tracking branch 'upstream/master' into undermine

Ioney committed:
formatting

Ioney committed:
undo ruRU formatting

tfmonk commented on plugins/11_TheWarWithin/zones/undermine.lua:
```diff
@@ -694,6 +782,137 @@ map.nodes[67482929] = ScrapHeap({criteriaID = 103187}) -- Blackwater Marina
+    elseif ns.PlayerHasItem(235053) then
+        local function ChettStatus()
+            local quests = {
+                {86915, L['side_with_a_cartel']}, {86917, L['ship_right']},
```
If we have the quest ids we shouldn't need a localization string for the quest name it could just be pulled

pepedressingroom replied:
It seems that these quests do not have localization strings returned by `C_QuestLog.GetTitleForQuestID`, only English.
I am not sure if there's any other API can return localization strings.

tfmonk replied:
are you checking on the English client for the string or are you swapping your game language over? as you wouldn't see the localised versions for other languages with the English client... you need to physically swap the game language over for it to return the localised string

pepedressingroom replied:
I only get the English version strings on the zhCN client, but I didn't try the other languages.
Maybe these quests are special and need the other API to get the localized strings?

tfmonk replied:
That might be the case but I would assume that if they were quests it should follow the same logic as other ones. Do they show up in English on the chest list in the zhCN client or are they localised?

Dathwada replied:
The german client also returns the english quest title `Side with a Cartel` for `C_QuestLog.GetTitleForQuestID(86915)`

pepedressingroom replied:
These quest have localised quest names and descriptions when I open C.H.E.T.T. , I wrote down all the strings using  several alts.
But when I tried to use `/dump C_QuestLog.GetTitleForQuestID(86915)` in chat or embed `‘{quest:86915}’` in lua, the quests all return English, so that I guess other non-English clients also show English.

Krovikan replied:
In CriteriaTree.db2 is all localized. Thanks for @adavak tip, below.

Regards

tfmonk commented on plugins/11_TheWarWithin/zones/isle_of_dorn.lua:
```diff
@@ -658,6 +659,14 @@ dor.nodes[61561850] =
     PT.Tailoring({quest = 83922, id = 226348, parent = map.id}) -- Dornogal Seam Ripper
 map.nodes[56216102] = PT.Tailoring({quest = 83923, id = 226349}) -- Earthen Tape Measure
 
+dor.nodes[49295561] = Vendor({
```
This node could do with a profession dependency on it to only show it for players with engineering

pepedressingroom replied:
This item do not need Engineering to learn. 
I bought and learned this item with a non-engineer character who had looted the treasure, but missed this item.

> If you've looted the treasure before and haven't gotten the recipe, you can also purchase it from engineering supply vendors like [Supply Foreman Drezmol](https://www.wowhead.com/npc=219098) as a backup. Note that the NPC will only sell [Crumpled Schematic: Wormhole Generator: Undermine](https://www.wowhead.com/item=235037) if you've looted the [Crumpled Schematic](https://www.wowhead.com/object=504096) treasure and didn't obtain the recipe due to some bug.

tfmonk replied:
So it's an account wide unlock then? I only mentioned the engineering dependency as that's what the wormhole toy needs to use it or is that only to learn it the initial time then all characters can use it?

pepedressingroom replied:
Yes, it is unlocked account wide. 
My non-engineer mage learned it, and engineer rogue have the option in the toy.

Krovikan replied:
Unlocked yes, but they can use without engineering?

Regards

pepedressingroom replied:
Yes, Krovikan. No need to be an engineer to use this item.

tfmonk replied:
ok so in that case, I just tested it and yeah the schematic anyone can learn and only the characters with eng get the benefits of it and looks like its triggers a quest 86630 and you have that in there and it hides after learning it, so I'm happy enough with this node now

pepedressingroom committed:
display name of C.H.E.T.T. List

pepedressingroom committed:
Merge remote-tracking branch 'origin/undermine' into undermine

adavak commented:
![图片](https://github.com/user-attachments/assets/b04e464f-688d-4e08-bbe3-42e62618dea8)

all about this non-localization, is from QuestV2CliTask.db2

but in CriteriaTree.db2 the same name was localized.

Krovikan committed:
Updated Undermine esES

Krovikan committed:
Fixed chett_extra 40x enUS

Krovikan commented:
@pepedressingroom The quantities of someone (currency or items) have this format: `40x {currency:XXXXXX}`

I added the `x` to your line 657:
`L['chett_extra'] = 'Exchange 40x {item:236668} for an extra {item:236682}.\n'`

I have other petition. I need help to understand this lines:

```
652 L['chett_exchange_1'] = 'Chose to exchange {item:235053} for?\n\n #1. ' .. ns.color.LightBlue('250') .. ' {currency:3008}\n (Bring back to {npc:238029}).\n\n #2. ' -- REVIEW

653 L['chett_exchange_2a'] = ns.color.LightBlue('500') .. ' {currency:%d} Reputation\n (Buy {item:%d} from {faction:%d} Quartermaster)\n' --REVIEW

654 L['chett_exchange_2b'] = ns.color.LightBlue('500') .. ' Aligned Cartel Reputation\n ( {note:Have to side with a cartel first this week!})\n' -- REVIEW
```

I don't know how to translate with all this code inside. Remember the `x` for valors in theses lines (250 and 500).

Maybe I can understand good these 3 lines if you can show me how is seen in the game (screenshots).

Regards

pepedressingroom commented:
@Krovikan 
If you have sided with a cartel this week, option #2  will look like this.
`L['chett_exchange_1'] .. L['chett_exchange_2a']`

> Chose to exchange Completed C.H.E.T.T. List for?
> 
>  #1.  250 Valorstones
>  (Bring back to {npc:238029}).
>  
>   #2.  500 {currency:%d} Reputation
>  (Buy {item:%d} from {faction:%d} Quartermaster)

If you haven't side with a cartel this week, option #2  will look like this.
`L['chett_exchange_1'] .. L['chett_exchange_2b']`

> Chose to exchange Completed C.H.E.T.T. List for?
> 
>  #1.  250 Valorstones
>  (Bring back to {npc:238029}).
> 
>  #2. 500 Aligned Cartel Reputation
> ( {note:Have to side with a cartel first this week!})

I am sorry I made it too complex to understand.

Krovikan committed:
Updated Undermine esES

Krovikan commented:
Thanks for explanations.

> 500 {currency:%d} Reputation
> 250 Valorstones

Only one last thing. The quoted text needs to be:
500`x` {currency:%d} Reputation
250`x` Valorstones

~~I don't know if you need to do in code or in locales. But better do it yourself.~~

I saw the light. The ' ' is string, so I added the `x` to enUS.lua and esES.lua.

Regards

Krovikan committed:
Updated 'x' esES

Krovikan committed:
Updated 'x' enUS

pepedressingroom committed:
translation

pepedressingroom committed:
Merge remote-tracking branch 'origin/undermine' into undermine

pepedressingroom committed:
D.R.I.V.E. Customizations

fix an error when using GetAchievementInfo
add L['rep_paragon'] description for vendors
add a  checklist for D.R.I.V.E. customizations

pepedressingroom commented:
```
Rares = { -- (Daily)
-- + 20 {faction:2653}
84917, -- Scrapbeak
84918, -- Ratspit
84919, -- Tally Doublespeak
84920, -- V.V. Goosworth
84921, -- Thwack
84922, -- S.A.L.
84926, -- Nitro
84927, -- Candy Stickemup
84928, -- Grimewick
85004, -- Swigs Farsight
87007, -- Gallagio Garbage
}
```
```
RareElites = { -- (Weekly)
-- + 50 {faction:2653}
84877, -- Ephemeral Agent Lathyd
84884, -- The Junk-Wall
84895, -- Slugger the Smart
84907, -- Chief Foreman Gutso
84911, -- Flyboy Snooty
}
```
```
SummonedRareElites = { -- (Weekly)
-- + 100 {faction:2653}
-- + 150 {faction:alignedCartel}
{90491, 85778, 86431}, -- Scrapchewer
{90488, 86298, 86414}, -- M.A.G.N.O.
{90489, 86307, 86390}, -- Giovante
{90490, 85777, 86428}, -- Voltstrike the Charged
{90492, 85010, 85013}, -- Darkfuse Precipitant
}
-- SummonedRareElites[i][1] (9xxxx) have enUS strings returned by API
-- SummonedRareElites[i][2] (8xxxx) are hidden
-- SummonedRareElites[i][3] (8xxxx) have localised quest name , type of which is "Bonus Ojective with Completion Toast". These quests are not related with the loot tables.
```
To merge and resolve the rares quest conflicts, I have checked my QuestsChanged saved variables.

I looted "Darkfuse Precipitant" in Week 2 (since UM patch) with only `quest:90492` flagged, and I remembered I did not receive any reputation.
But in Week 3, I triggered both `quest:90492` and `quest:85010`, probably because I unlocked "Darkfuse Precipitant" in that week.
Darkfuse Precipitant rewards reputation now. Maybe `quest:85010` was unlocked by the Darkfuse quest line, or Blizzard hotfixed it.

So that I use `SummonedRareElites[i][2]` (8xxxx) instead of `SummonedRareElites[i][1]` (9xxxx) for reputation quest ids and use `SummonedRareElites[i][1]` (9xxxx) for normal loot tables.

> FYI:
> [Lazey's comment](https://www.wowhead.com/achievement=41216/adventurer-of-undermine#comments) on 2025/02/13

Regards

adavak commented:
i tested 8xxxx quest id worked perfect. do not need switch 9xxxx.

pepedressingroom commented:
> i tested 8xxxx quest id worked perfect. do not need switch 9xxxx.

Yes, they are twins quests and interchangeable most of time.

tfmonk commented:
> > i tested 8xxxx quest id worked perfect. do not need switch 9xxxx.
> 
> Yes, they are twins quests and interchangeable most of time.

Be careful with them tho, tracking the wrong quest could cause them to display as complete or not complete(even when killed) on different characters, when two quests are triggered, ideally you should then validate which is then triggered on a different char and that would be the rare kill tracking quest the other should be the rep

adavak commented:
> > > i tested 8xxxx quest id worked perfect. do not need switch 9xxxx.
> > 
> > 
> > Yes, they are twins quests and interchangeable most of time.
> 
> Be careful with them tho, tracking the wrong quest could cause them to display as complete or not complete(even when killed) on different characters, when two quests are triggered, ideally you should then validate which is then triggered on a different char and that would be the rare kill tracking quest the other should be the rep

The existing questid has been verified. I used 3 characters and killed them every day without missing any.

pepedressingroom committed:
Merge remote-tracking branch 'upstream/master' into undermine

pepedressingroom committed:
only 3 WQs are locked originally

pepedressingroom committed:
Merge remote-tracking branch 'upstream/master' into undermine

pepedressingroom committed:
aligned cartel reputation rewards

pepedressingroom committed:
format

pepedressingroom committed:
format

adavak committed:
lint

pepedressingroom committed:
reputation quests changes

pepedressingroom committed:
Merge remote-tracking branch 'origin/undermine' into undermine

pepedressingroom commented:
@adavak @tfmonk 
Blizzard has removed `SummonedRareElites[i][2]` (8xxxx) in patch 11.1.5, 
and changes reputation quests  to `SummonedRareElites[i][1]` (9xxxx)

```
SummonedRareElites = { -- (Weekly)
-- + 100 {faction:2653}
-- + 300 {faction:alignedCartel} 
{90491, nil}, -- Scrapchewer
{90488, nil}, -- M.A.G.N.O.
{90489, nil}, -- Giovante
{90490, nil}, -- Voltstrike the Charged
{90492, nil}, -- Darkfuse Precipitant
}
```

adavak commented:
> @adavak @tfmonk Blizzard has removed `SummonedRareElites[i][2]` (8xxxx) in patch 11.1.5, and changes reputation quests to `SummonedRareElites[i][1]` (9xxxx)
> 
> ```
> SummonedRareElites = { -- (Weekly)
> -- + 100 {faction:2653}
> -- + 300 {faction:alignedCartel} 
> {90491, nil}, -- Scrapchewer
> {90488, nil}, -- M.A.G.N.O.
> {90489, nil}, -- Giovante
> {90490, nil}, -- Voltstrike the Charged
> {90492, nil}, -- Darkfuse Precipitant
> }
> ```

thanks, i got that, after map icon clean and login alts the purple icon show up. i know the questid changed.

pepedressingroom commented:
> > @adavak @tfmonk Blizzard has removed `SummonedRareElites[i][2]` (8xxxx) in patch 11.1.5, and changes reputation quests to `SummonedRareElites[i][1]` (9xxxx)
> > ```
> > SummonedRareElites = { -- (Weekly)
> > -- + 100 {faction:2653}
> > -- + 300 {faction:alignedCartel} 
> > {90491, nil}, -- Scrapchewer
> > {90488, nil}, -- M.A.G.N.O.
> > {90489, nil}, -- Giovante
> > {90490, nil}, -- Voltstrike the Charged
> > {90492, nil}, -- Darkfuse Precipitant
> > }
> > ```
> 
> thanks, i got that, after map icon clean and login alts the purple icon show up. i know the questid changed.

Due to this change, I guess you can receive reputation as a raid member now.
But there may be a bad situation. 
If you kill these 5 rares before aligning with a Cartel, you will miss out on your aligned Cartel's reputation for the week.
If it was the case, it's better to add a warning in the notes.
But I cannot test it this week on live realm.

regards

adavak committed:
Merge branch 'master' into undermine

adavak commented on plugins/11_TheWarWithin/zones/undermine.lua:
```diff
@@ -711,6 +804,171 @@ map.nodes[67482929] = ScrapHeap({criteriaID = 103187}) -- Blackwater Marina
+    return C_Spell.GetSpellDescription(1222389)
+end
+
+map.nodes[15852496] = ShippingHandling()
```
`map.nodes[90003500] = ShippingHandling()`

better pos, some addons will cover it.

adavak commented on plugins/11_TheWarWithin/zones/undermine.lua:
```diff
@@ -711,6 +804,171 @@ map.nodes[67482929] = ScrapHeap({criteriaID = 103187}) -- Blackwater Marina
+-------------------------- D.R.I.V.E. CUSTOMIZATIONS --------------------------
+-------------------------------------------------------------------------------
+
+map.nodes[15852996] = Collectible({
```
`map.nodes[90004000] = Collectible({`

better pos, some addons will cover it.

adavak commented on plugins/11_TheWarWithin/zones/undermine.lua:
```diff
@@ -711,6 +804,171 @@ map.nodes[67482929] = ScrapHeap({criteriaID = 103187}) -- Blackwater Marina
+    return note
+end
+
+map.nodes[15851996] = ChettList()
```
`map.nodes[90003000] = ChettList()`

better pos, some addons will cover it.

pepedressingroom committed:
better pos

pepedressingroom committed:
Merge remote-tracking branch 'origin/undermine' into undermine

pepedressingroom committed:
fix
